### PR TITLE
Use Xvfb for testing if $DISPLAY is not set

### PR DESCRIPTION
**Pull request by fhs**
This should enable tests to run in Travis CI. A fake devdraw can be used later when we have it.

Update #64

### CODE REVIEW

**Comment by sbinet** [comment]
Only tangential to the issue at hand, but somebody implemented 'devdraw' on top of 'exp/shiny'.

I suspect it would be easier to leverage for tests?

It's over there:

https://github.com/ktye/duitdraw/blob/master/README.md

File: fsys_test.go [hunk 1]
```diff
@@ -12,17 +14,62 @@ import (
...
+	panic("no free X server number")
+}
+
 func startAcme(t *testing.T) (*exec.Cmd, *client.Fsys) {
 	// Fork off an acme and talk with it.
 
-	os.Setenv("NAMESPACE", os.TempDir()+"/ns.fsystest")
-	os.Mkdir(os.TempDir()+"/ns.fsystest", os.ModeDir|os.ModePerm)
-	os.Remove(os.TempDir() + "/ns.fsystest/acme")
+	ns := os.TempDir() + "/ns.fsystest"
```
Comment (rjkroege) [inline fsys_test.go]: I'd have a preference for `filepath.Join` here and elsewhere and `ioutil.TempDir`  if you're willing to update the PR. (Yes, flux should have probably written it this way in the first place. 😊)

Comment (rjkroege) [reply fsys_test.go]: But I see that you already did that. My apologies. I was confused by the github interface.

File: fsys_test.go [hunk 2]
```diff
@@ -12,35 +15,100 @@ import (
...
+		case "linux", "freebsd", "openbsd", "netbsd", "dragonfly":
+			if os.Getenv("DISPLAY") == "" {
+				dp := fmt.Sprintf(":%d", xvfbServerNumber())
+				x = exec.Command("Xvfb", dp)
+				if err := x.Start(); err != nil {
+					log.Fatalf("failed to execute Xvfb: %v", err)
+				}
+				// Give Xvfb some time to start up.
+				// 3 seconds is default for xvfb-run.
+				time.Sleep(3 * time.Second)
```
Comment (rjkroege) [inline fsys_test.go]: Do you have any experience with how flaky Travis will be with that?

Comment (fhs) [reply fsys_test.go]: It's also 3 seconds in the [Travis CI docs](https://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-xvfb-directly), so I doubt it'll be an issue. If it really becomes necessary, we can do a loop until `xdpyinfo` succeeds, but that would be another external dependency.

**Pull request merged**